commit c5d9339a62760c6e11b0a7f9927db117d2ef5351
Author: Eamon Walsh <ewalsh@termysequence.com>
Date:   Fri Jan 18 03:12:34 2019 -0500

    qt: Set clipboard selection text following all selection edits (Issue #5)

diff --git a/src/base/selection.cpp b/src/base/selection.cpp
index d14d10c..c6544d4 100644
--- a/src/base/selection.cpp
+++ b/src/base/selection.cpp
@@ -52,6 +52,7 @@ Selection::reportResult()
 {
     if (checkResult()) {
         hasAnchor = true;
+        clipboardSelect();
         m_buffers->activateSelection();
         emit activated();
     } else {
@@ -61,6 +62,14 @@ Selection::reportResult()
     }
 }
 
+inline void
+Selection::reportModified()
+{
+    clipboardSelect();
+    m_buffers->reportRegionChanged();
+    emit modified();
+}
+
 void
 Selection::selectBuffer()
 {
@@ -284,9 +293,7 @@ Selection::forwardWord(bool upper)
 
     anchorRow = *row;
     rc = true;
-    // Assumed active, no check
-    m_buffers->reportRegionChanged();
-    emit modified();
+    reportModified(); // Assumed active
 out:
     return std::make_pair(*row, rc);
 }
@@ -331,9 +338,7 @@ Selection::backWord(bool upper)
 
     anchorRow = *row;
     rc = true;
-    // Assumed active, no check
-    m_buffers->reportRegionChanged();
-    emit modified();
+    reportModified(); // Assumed active
 out:
     return std::make_pair(*row, rc);
 }
@@ -359,9 +364,7 @@ Selection::forwardChar(bool upper)
         anchorRow = *row;
         anchorCol = ++*col;
         rc = true;
-        // Assumed active, no check
-        m_buffers->reportRegionChanged();
-        emit modified();
+        reportModified(); // Assumed active
     }
 out:
     return std::make_pair(*row, rc);
@@ -388,9 +391,7 @@ Selection::backChar(bool upper)
         anchorRow = *row;
         anchorCol = --*col;
         rc = true;
-        // Assumed active, no check
-        m_buffers->reportRegionChanged();
-        emit modified();
+        reportModified(); // Assumed active
     }
 out:
     return std::make_pair(*row, rc);
@@ -419,9 +420,7 @@ Selection::forwardLine(bool upper)
     anchorRow = ++*row;
     anchorCol = *col = next;
     rc = true;
-    // Assumed active, no check
-    m_buffers->reportRegionChanged();
-    emit modified();
+    reportModified(); // Assumed active
 out:
     return std::make_pair(*row, rc);
 }
@@ -449,9 +448,7 @@ Selection::backLine(bool upper)
     anchorRow = --*row;
     anchorCol = *col = next;
     rc = true;
-    // Assumed active, no check
-    m_buffers->reportRegionChanged();
-    emit modified();
+    reportModified(); // Assumed active
 out:
     return std::make_pair(*row, rc);
 }
diff --git a/src/base/selection.h b/src/base/selection.h
index 70f1898..d301a6b 100644
--- a/src/base/selection.h
+++ b/src/base/selection.h
@@ -29,6 +29,7 @@ private:
 
     bool checkResult();
     void reportResult();
+    void reportModified();
 
 signals:
     void activated();
